### PR TITLE
fix TokenToBuyList weirdness when favoriting

### PR DESCRIPTION
**Pull request by greg-schrammel**
Fixes APP-####

## Screen recordings / screenshots

before/after

https://github.com/user-attachments/assets/5883c2f2-c008-4df2-9918-6c2eb835817d

https://github.com/user-attachments/assets/57bfb30a-43ed-4431-9661-f2444a57d830

### CODE REVIEW

File: src/components/asset-list/RecyclerAssetList2/WrappedCollectiblesHeader.tsx
Comment (walmat) [inline src/components/asset-list/RecyclerAssetList2/WrappedCollectiblesHeader.tsx:1]: why was this file changed?

File: src/__swaps__/screens/Swap/components/TokenList/TokenToBuyList.tsx [hunk 1]
```diff
@@ -144,7 +148,7 @@ export const TokenToBuyList = () => {
         ListHeaderComponent={<ChainSelection output />}
         contentContainerStyle={{ paddingBottom: 16 }}
         // For some reason shallow copying the list data allows FlashList to more quickly pick up changes
```
Comment (walmat) [inline src/__swaps__/screens/Swap/components/TokenList/TokenToBuyList.tsx:150]:
```suggestion
```
rm this comment if we're removing the `.slice()`

File: src/__swaps__/screens/Swap/components/TokenList/TokenToBuyList.tsx [hunk 2]
```diff
@@ -72,6 +73,16 @@ export type HeaderItem = { listItemType: 'header'; id: AssetToBuySectionId; data
...
+const ScrollViewWithRef = forwardRef<Animated.ScrollView>(function ScrollViewWithRef(props, ref) {
+  const { outputProgress } = useSwapContext();
+  const animatedListProps = useAnimatedProps(() => {
+    const isFocused = outputProgress.value === 2;
+    return { scrollIndicatorInsets: { bottom: 28 + (isFocused ? EXPANDED_INPUT_HEIGHT - FOCUSED_INPUT_HEIGHT : 0) } };
+  });
+  // eslint-disable-next-line react/jsx-props-no-spreading
+  return <Animated.ScrollView {...props} ref={ref} animatedProps={animatedListProps} />;
+});
+
```
Comment (walmat) [inline src/__swaps__/screens/Swap/components/TokenList/TokenToBuyList.tsx:85]: mostly a question for myself: "what happens for the input token list?"

File: src/components/asset-list/RecyclerAssetList2/WrappedCollectiblesHeader.tsx
Comment (greg-schrammel) [reply src/components/asset-list/RecyclerAssetList2/WrappedCollectiblesHeader.tsx:1]: ops

File: src/__swaps__/screens/Swap/components/TokenList/TokenToBuyList.tsx (hunk 2 above)
Comment (walmat) [reply src/__swaps__/screens/Swap/components/TokenList/TokenToBuyList.tsx:85]: seems unaffected, so we're good here

**Review by walmat — APPROVED** [review]
